packages/devextreme/js/__internal/grids/grid_core/sticky_columns/m_sticky_columns.ts

```diff
@@ -69,7 +69,70 @@ const baseStickyColumns = <T extends ModuleType<ColumnsView>>(Base: T) => class
 
 const columnHeadersView = (
   Base: ModuleType<ColumnHeadersView>,
-) => class ColumnHeadersViewStickyColumnsExtender extends baseStickyColumns(Base) {};
+) => class ColumnHeadersViewStickyColumnsExtender extends baseStickyColumns(Base) {
+  public getContextMenuItems(options) {
+    const { column } = options;
+    const columnFixingOptions: any = this.option('columnFixing');
+    let items: any = super.getContextMenuItems(options);
+
+    if (options.row && options.row.rowType === 'header') {
+      if (columnFixingOptions.enabled === true && column && column.allowFixing) {
+        const onItemClick = (params) => {
+          // eslint-disable-next-line default-case
+          switch (params.itemData.value) {
+            case 'none':
+              this._columnsController.columnOption(column.index, 'fixed', false);
+              break;
+            case 'left':
+              this._columnsController.columnOption(column.index, {
+                fixed: true,
+                fixedPosition: 'left',
+              });
+              break;
+            case 'right':
+              this._columnsController.columnOption(column.index, {
+                fixed: true,
+                fixedPosition: 'right',
+              });
+              break;
+            case 'sticky':
+              this._columnsController.columnOption(column.index, {
+                fixed: true,
+                fixedPosition: 'sticky',
+              });
+              break;
+          }
+        };
+
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        items = items || [];
+        items.push(
+          {
+            text: columnFixingOptions.texts.fix,
+            beginGroup: true,
+            items: [
+              {
+                text: columnFixingOptions.texts.leftPosition, value: 'left', disabled: column.fixed && (!column.fixedPosition || column.fixedPosition === 'left'), onItemClick,
+              },
+              {
+                text: columnFixingOptions.texts.rightPosition, value: 'right', disabled: column.fixed && column.fixedPosition === 'right', onItemClick,
+              }],
+          },
+          {
+            text: columnFixingOptions.texts.unfix, value: 'none', disabled: !column.fixed, onItemClick,
+          },
+        );
+
+        if (this._isStickyColumns()) {
+          items[items.length - 2].items.push({
```

Comment:
It is unclear code and unclear structure. I'd suggest to create items first and then to add them. For example:
```
const fixOptions = [{ ... }, { ... }];
if(this._isStickyColumns()) {
   fixOptions.push { ... value: 'sticky' ... }
}
items.push(fixOptions);
items.push(unfixOptions);
```